fiftyone/utils/eval/segmentation.py

```diff
@@ -495,21 +513,34 @@ def _from_dict(cls, d, samples, config, eval_key, **kwargs):
         )
 
     @staticmethod
-    def _parse_confusion_matrix(confusion_matrix, classes):
+    def _parse_confusion_matrix(confusion_matrix, classes, ytrue_ids_dict, ypred_ids_dict):
         ytrue = []
         ypred = []
         weights = []
+        ytrue_ids = None
+        ypred_ids = None
+        if ytrue_ids_dict is not None and ypred_ids_dict is not None:
+            ytrue_ids = []
+            ypred_ids = []
         nrows, ncols = confusion_matrix.shape
         for i in range(nrows):
             for j in range(ncols):
+                index = (i, j)
                 cij = confusion_matrix[i, j]
                 if cij > 0:
-                    ytrue.append(classes[i])
-                    ypred.append(classes[j])
-                    weights.append(cij)
-
-        return ytrue, ypred, weights
-
+                    if ytrue_ids_dict is not None and ypred_ids_dict is not None:
```

Comment:
Minor, maybe create a bool flag for this as this is re-used above in line 522 and also in every cell in this doubly-nested loop

Reply:
Fixed in latest commit